kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -0,0 +1,970 @@
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild80211WirelessSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenNoExceptionsHaveBeenThrown();
+        thenResultingMapContains("mode", "ap");
+        thenResultingMapContainsBytes("ssid", "ssidtest");
+        thenResultingMapContains("band", "a");
+        thenResultingMapContains("channel", new UInt32(Short.parseShort("10")));
+    }
+
+    @Test
+    public void build80211WirelessSettingsShouldWorkWhenGivenExpectedMapAndModeMalformed() {
+
+        givenMapWith("net.interface.wlan0.config.wifi.mode", "MALFORMED_VALUE");
+        givenMapWith("net.interface.wlan0.config.wifi.master.ssid", "ssidtest");
+        givenMapWith("net.interface.wlan0.config.wifi.master.radioMode", "RADIO_MODE_80211a");
+        givenMapWith("net.interface.wlan0.config.wifi.master.channel", "10");
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild80211WirelessSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenIllegalArgumentExceptionThrown();
+    }
+
+    @Test
+    public void build80211WirelessSettingsShouldWorkWhenGivenExpectedMapAndBandBg() {
+
+        givenMapWith("net.interface.wlan0.config.wifi.mode", "INFRA");
+        givenMapWith("net.interface.wlan0.config.wifi.infra.ssid", "ssidtest");
+        givenMapWith("net.interface.wlan0.config.wifi.infra.radioMode", "RADIO_MODE_80211b");
+        givenMapWith("net.interface.wlan0.config.wifi.infra.channel", "10");
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild80211WirelessSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenNoExceptionsHaveBeenThrown();
+        thenResultingMapContains("mode", "infrastructure");
+        thenResultingMapContainsBytes("ssid", "ssidtest");
+        thenResultingMapContains("band", "bg");
+        thenResultingMapContains("channel", new UInt32(Short.parseShort("10")));
+    }
+
+    @Test
+    public void build80211WirelessSettingsShouldWorkWhenGivenExpectedMapAndBandBgExt() {
+
+        givenMapWith("net.interface.wlan0.config.wifi.mode", "INFRA");
+        givenMapWith("net.interface.wlan0.config.wifi.infra.ssid", "ssidtest");
+        givenMapWith("net.interface.wlan0.config.wifi.infra.radioMode", "RADIO_MODE_80211nHT20");
+        givenMapWith("net.interface.wlan0.config.wifi.infra.channel", "10");
+        givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+
+        whenBuild80211WirelessSettingsIsRunWith(this.networkProperties, "wlan0");
+
+        thenNoExceptionsHaveBeenThrown();
+        thenResultingMapContains("mode", "infrastructure");
+        thenResultingMapContainsBytes("ssid", "ssidtest");
+        thenResultingMapContains("band", "bg");
+        thenResultingMapContains("channel", new UInt32(Short.parseShort("10")));
```

Comment:
Again unnecessary complication:

```suggestion
        thenResultingMapContains("channel", new UInt32(10));
```